File: service/service/place/place-api/src/main/java/bs/service/place/api/service/PlaceService.java
package bs.service.place.api.service;

import bs.lib.common.model.generated.LookupResultSet;
import bs.lib.common.model.generated.NewRecordVTO;
import bs.lib.sql.db.adapter.model.generated.OrderDirections;
import bs.service.place.model.generated.PlaceDTO;
import bs.service.place.model.generated.PlaceResultSet;
import bs.service.place.model.generated.PlaceVTO;

import java.time.LocalDate;

public interface PlaceService {
    NewRecordVTO createPlace(PlaceDTO placeDTO);
    NewRecordVTO updatePlace(Integer placeId, PlaceDTO placeDTO);
    void deletePlaceById(Integer placeId);
    PlaceVTO getPlaceById(Integer placeId, LocalDate createdOnFrom,LocalDate createdOnTo);
    PlaceResultSet getAllPlacesByFilter(String quickSearch, Integer pageNum, Integer pageSize,
                                        OrderDirections orderDir, String orderBy);
    LookupResultSet getAllPlacesLookup();
}

File: olympic-academy-working/src/app/core/services/place.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { PlaceDTO, PlaceVTO, PlaceResultSet } from '../models/place.model';
import { NewRecordVTO, LookupResultSet } from '../models/common.model';

@Injectable({
  providedIn: 'root'
})
export class PlaceService {
  constructor(private api: ApiService) {}

  // Lookup
  getAllPlacesLookup(): Observable<LookupResultSet> {
    return this.api.get('/lookup/places');
  }

  // CRUD
  createPlace(data: PlaceDTO): Observable<NewRecordVTO> {
    return this.api.post('/places', data);
  }

  getAllPlacesByFilter(params?: any): Observable<PlaceResultSet> {
    return this.api.get('/places', params);
  }

  getAllPlacesDetailsByFilter(params?: any): Observable<PlaceResultSet> {
    return this.api.get('/places/details', params);
  }

  getPlaceById(id: number): Observable<PlaceVTO> {
    return this.api.get(`/places/${id}`);
  }

  updatePlace(id: number, data: PlaceDTO): Observable<NewRecordVTO> {
    return this.api.put(`/places/${id}`, data);
  }

  deletePlace(id: number): Observable<void> {
    return this.api.delete(`/places/${id}`);
  }
}